### DISCUSSION TITLE
A way to enable transformGifs, but disable it for listing in entries and assets, which is causing timeouts

### DISCUSSION THREAD

**Discussion by arifje** [craftcms/cms#14613] [Ideas]
We're developing a website that is mainly focused on publishing GIF images so we need the width & height of the GIF assets to get everything working properly. If I am correct the only way of storing this data is to set **transformGifs** to **true**. 

This works fine during uploading of the images (imported via FeedMe btw), because the images are processed via queue job in the background, but when we edit an entry or try to browse through the assets section, we're getting server timeouts and browsers freezes because all GIF files are getting listed (and checked/processed again?).

When I set **transformGifs** to **false** the listing works fine, and even small thumbnails are generated of the GIFs.

Because we need the width & height a solution would be of we could set transformGifs to true, but somehow get it to false in case of listing in an asset field when editing the entry in the CP, but also browsing through the assets sections.

PS. I also tried using the plugin [metadata](https://github.com/carlcs/craft-assetmetadata/), which seems to grab the dimensions also, but unfortunately this is not triggered when assets are imported with FeedMe.

**Comment by brandonkelly** [comment]
> If I am correct the only way of storing this data is to set **transformGifs** to **true**.

Craft will record the original image width/height on upload, regardless of whether it needs to transform it.

**Comment by arifje** [comment]
Thanks for the reply @brandonkelly. 

I think I drew my conclusion to soon on this, because I now see dimensions are set after upload. But, the reason is we've imported 50k GIF with FeedMe and for some reasons 30k+ of assets don't have any width/height:

![Screenshot 2024-03-18 at 13 12 46](https://github.com/craftcms/cms/assets/1286410/10829f35-a524-4394-a30f-02322c001c6e)

<img width="459" alt="Screenshot 2024-03-19 at 07 33 22" src="https://github.com/craftcms/cms/assets/1286410/34c9dfbd-eb0c-480f-900c-5ff8edfe36a0">


I had contact with support and Oli advised me to run a re-index, so I did. But this does not seems to fix it. I wrote I little Twig extension that uses **getimagesize()** to check if the files return a valid dimension, and they did.

Maybe I am overseeing something here?

**Reply by brandonkelly** [reply]
Strange, let’s keep digging from support.
